demo/top_down_video_demo_full_frame_without_det_gpuaccel.py

```diff
@@ -0,0 +1,207 @@
+    assert frame_resize.shape[1::-1] == tuple(
+        img_metas['ann_info']['image_size'])
+    frame_cuda = torch.from_numpy(frame_resize).to(device).float()
+    frame_cuda = frame_cuda.permute(2, 0, 1)  # HWC to CHW
+    mean = torch.from_numpy(img_metas['img_norm_cfg']['mean']).to(device)
+    std = torch.from_numpy(img_metas['img_norm_cfg']['std']).to(device)
+    frame_cuda = F.normalize(frame_cuda, mean=mean, std=std, inplace=True)
+    frame_cuda = frame_cuda[None, :, :, :]  # NCHW
+    data = {'img': frame_cuda, 'img_metas': [img_metas]}
+    return data
+
+
+def main():
+    """Visualize the demo video with GPU acceleration.
+
+    Using mmdet to detect the human.
+    """
+    parser = ArgumentParser()
+    parser.add_argument('pose_config', help='Config file for pose')
+    parser.add_argument('pose_checkpoint', help='Checkpoint file for pose')
+    parser.add_argument('--video-path', type=str, help='Video path')
+    parser.add_argument(
+        '--show',
+        action='store_true',
+        default=False,
+        help='whether to show visualizations.')
+    parser.add_argument(
+        '--out-video-root',
+        default='',
+        help='Root of the output video file. '
+        'Default not saving the visualization video.')
+    parser.add_argument(
+        '--device', default='cuda:0', help='Device used for inference')
+    parser.add_argument(
+        '--nvdecode', action='store_true', help='Use NVIDIA decoder')
+    parser.add_argument(
+        '--kpt-thr', type=float, default=0.3, help='Keypoint score threshold')
+    parser.add_argument(
+        '--radius',
+        type=int,
+        default=4,
+        help='Keypoint radius for visualization')
+    parser.add_argument(
+        '--thickness',
+        type=int,
+        default=1,
+        help='Link thickness for visualization')
+
+    args = parser.parse_args()
+
+    assert args.show or (args.out_video_root != '')
+    # build the pose model from a config file and a checkpoint file
+    pose_model = init_pose_model(
+        args.pose_config, args.pose_checkpoint, device=args.device.lower())
+
+    dataset = pose_model.cfg.data['test']['type']
+    dataset_info = pose_model.cfg.data['test'].get('dataset_info', None)
+    if dataset_info is None:
+        warnings.warn(
+            'Please set `dataset_info` in the config.'
+            'Check https://github.com/open-mmlab/mmpose/pull/663 for details.',
+            DeprecationWarning)
+    else:
+        dataset_info = DatasetInfo(dataset_info)
+
+    pose_model.cfg.data['test']['dataset_info'] = dataset_info
+    if args.nvdecode:
+        VideoCapture = ffmpegcv.VideoCaptureNV
+    else:
+        VideoCapture = ffmpegcv.VideoCapture
+    video_origin = VideoCapture(args.video_path)
+    img_metas = prefetch_img_metas(pose_model.cfg,
+                                   (video_origin.width, video_origin.height))
+    resize_wh = pose_model.cfg.data_cfg['image_size']
+    video_resize = VideoCapture(
```

Comment:
Is this `resize` operation in video reading equivalent to the `TopDownAffine` pipeline in MMPose?

Reply:
Yes, the `resize` is a short cut version of `TopDownAffine`. In the video inference, most people don't need 'flip', 'rotation' operations. In this case, **the `TopDownAffine` can be degenerated as video resize and padding**, which is implemented by high performance ffmpeg now.